### PR TITLE
Migrating `SourceCodeEditor` to TS.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is migrating the `SourceCodeEditor` component to TS.

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/SourceCodeEditor.tsx [hunk 1]
```diff
@@ -297,11 +332,14 @@ class SourceCodeEditor extends React.Component {
                                $resizable={resizable}>
             <AceEditor ref={(c) => {
               this.reactAce = c;
+              // @ts-ignore
               if (innerRef) { innerRef.current = c; }
             }}
                        annotations={annotations}
+                       // @ts-expect-error
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:339]: It looks like we can avoid `@ts-expect-error` here by changing `$blockScrolling: 'Infinity'` to `$blockScrolling: Infinity`

File: graylog2-web-interface/src/components/common/SourceCodeEditor.tsx [hunk 2]
```diff
@@ -297,11 +332,14 @@ class SourceCodeEditor extends React.Component {
                                $resizable={resizable}>
             <AceEditor ref={(c) => {
               this.reactAce = c;
+              // @ts-ignore
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:335]: We can avoid `@ts-ignore` here by typing `innerRef` as a `MutableRefObject`.

File: graylog2-web-interface/src/components/common/SourceCodeEditor.tsx [hunk 3]
```diff
@@ -297,11 +332,14 @@ class SourceCodeEditor extends React.Component {
...
               if (innerRef) { innerRef.current = c; }
             }}
                        annotations={annotations}
+                       // @ts-expect-error
                        editorProps={{ $blockScrolling: 'Infinity' }}
                        // Convert Windows line breaks to Unix. See issue #7889
+                       // @ts-expect-error
                        setOptions={{ newLineMode: 'unix' }}
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:343]: I also had look at this error, but it looks like `unix` is the correct value and the types are wrongly expecting a boolean.

Comment (dennisoelkers) [reply graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:343]: Yes, there is even an issue for that: https://github.com/securingsincity/react-ace/issues/752

**Review by linuspahl — APPROVED** [review]

**Pull request merged**